applications/main/subghz/scenes/subghz_scene_signal_settings.c

```diff
@@ -0,0 +1,184 @@
+        for(uint8_t i = 0; i < PROTOCOLS_COUNT i++) {
+            if(!strcmp(furi_string_get_cstr(tmp_string), protocols[i].name)) {
+                mode_count = protocols[i].mode_count;
+                if(flipper_format_read_uint32(fff_data_file, "CounterMode", &tmp_counter_mode, 1)) {
+                    counter_mode = (uint8_t)tmp_counter_mode;
+                } else {
+                    counter_mode = 0;
+                }
+            }
+        }
+    }
+    FURI_LOG_I(TAG, "Current CounterMode value %li", counter_mode);
+
+    furi_string_free(tmp_string);
+    flipper_format_file_close(fff_data_file);
+    flipper_format_free(fff_data_file);
+    furi_record_close(RECORD_STORAGE);
+
+    //Create and Enable/Disable variable_item_list depent from current CounterMode value
+    VariableItemList* variable_item_list = subghz->variable_item_list;
+    int32_t value_index;
+    VariableItem* item;
+
+    variable_item_list_set_enter_callback (variable_item_list,subghz_scene_signal_settings_variable_item_list_enter_callback,subghz);
+    
+    item = variable_item_list_add(
+        variable_item_list,
+        "Counter Mode",
+        mode_count,
+        subghz_scene_signal_settings_counter_mode_changed,
+        subghz);
+    value_index = value_index_int32(counter_mode, counter_mode_value, mode_count);
+
+    variable_item_set_current_value_index(item, value_index);
+    variable_item_set_current_value_text(item, counter_mode_text[value_index]);
+    variable_item_set_locked(
+        item, (counter_mode == 0xff), " Not available \n for this \nprotocol !");
+
+    // item = variable_item_list_add(
+    //     variable_item_list,
+    //     "Counter Value",
+    //     1,
+    //     NULL,
+    //     subghz);
+    // variable_item_set_current_value_text(item, "FFFFFF");
+
+    view_dispatcher_switch_to_view(subghz->view_dispatcher, SubGhzViewIdVariableItemList);
+}
+
+bool subghz_scene_signal_settings_on_event(void* context, SceneManagerEvent event) {
+    SubGhz* subghz = context;
+    if(event.type == SceneManagerEventTypeBack) {
+        scene_manager_previous_scene(subghz->scene_manager);
+        return true;
+    } else
+        return false;
+}
+
+void subghz_scene_signal_settings_on_exit(void* context) {
+    SubGhz* subghz = context;
+    const char* file_path = furi_string_get_cstr(subghz->file_path);
+
+    furi_assert(subghz);
+    furi_assert(file_path);
+
+    // if ConterMode was changed from 0xff then we must update or write new value to file
+    if(counter_mode != 0xff) {
+        Storage* storage = furi_record_open(RECORD_STORAGE);
+        FlipperFormat* fff_data_file = flipper_format_file_alloc(storage);
+
+        // check is the file available for update or write brand new value and to it
+        if(flipper_format_file_open_existing(fff_data_file, file_path)) {
+            if(flipper_format_update_uint32(fff_data_file, "CounterMode", &counter_mode, 1)) {
```

Comment:
Whole section below can be replaced with `flipper_format_insert_or_update_uint32`